sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -489,75 +507,153 @@ def _resolve_inputs_for_pipeline_component_jobs(self, jobs, base_path):
-                azureml_type=AzureMLResourceType.COMPONENT,
-            )
+        # compute binding to pipeline input is supported on node.
+        supported_fields = ["compute", "compute_name"]
+        for field_name in supported_fields:
+            val = try_get_non_arbitrary_attr_for_potential_attr_dict(node, field_name)
+            if isinstance(val, PipelineInput):
+                # Put binding string to field
+                setattr(node, field_name, val._data_binding())
+
+    @classmethod
+    def _set_default_display_name_for_anonymous_component_in_node(cls, node: BaseNode, default_name: str):
+        """Set default display name for anonymous component in a node.
+        If node._component is an anonymous component and without display name, set the default display name.
+        """
+        if not isinstance(node, BaseNode):
+            return
+        component = node._component
+        if isinstance(component, PipelineComponent):
+            return
+        # Set display name as node name
+        # TODO: the same anonymous component with different node name will have different anonymous hash
+        # as their display name will be different.
+        if (
+                isinstance(component, Component)
+                # check if component is anonymous and not created based on its id. We can't directly check
+                # node._component._is_anonymous as it will be set to True on component creation,
+                # which is later than this check
+                and not component.id
+                and not component.display_name
+        ):
+            component.display_name = default_name
+
+    @classmethod
+    def _resolve_compute_for_node(cls, node: BaseNode, resolver):
+        """Resolve compute for base node."""
+        if not isinstance(node._component, PipelineComponent):
+            # Resolve compute for other type
+            # Keep data binding expression as they are
+            if not is_data_binding_expression(node.compute):
+                # Get compute for each job
+                node.compute = resolver(node.compute, azureml_type=AzureMLResourceType.COMPUTE)
+            if has_attr_safe(node, "compute_name") and not is_data_binding_expression(node.compute_name):
+                node.compute_name = resolver(node.compute_name, azureml_type=AzureMLResourceType.COMPUTE)
+
+    @classmethod
+    def _divide_nodes_to_resolve_into_layers(
+        cls,
+        component: PipelineComponent,
+        extra_operations: List[Callable]
+    ):
+        """Divide nodes to resolve into layers. Deeper layer nodes will be resolved
+        first so that their resolution result can be used by upper layer nodes.
+        """
+        # add an empty layer to mark the end of the first layer
+        layers, cur_head_in_layer, cur_layer_i = [list(component.jobs.items()), []], 0, 0
+
+        while cur_layer_i < len(layers) and cur_head_in_layer < len(layers[cur_layer_i]):
```

Comment:
cur_layer_i -> cur_layer ?

Reply:
Name it in this way as it's an index instead of a "layer" (list of nodes)